packages/components/src/__actions__/Button/v3/_docs/Button--api-specification.mdx

```diff
@@ -0,0 +1,160 @@
+
+```tsx
+<Button onPress={e => submit(e)}>
+  Submit
+</Button>
+```
+
+React Aria&apos;s implementation of `onPress` provides better support for consistent touch events across device types, such mobile, desktop and tablet. You can read more about the development and reason behind this pattern [here](https://react-spectrum.adobe.com/blog/building-a-button-part-1.html#touch-interactions).
+
+
+### Button content and children
+
+Labels and any button content can be passed into the `Button` as `children`. Content wrapped by the `Button` will be spaced using `flex-gap` relative to the `size` prop. This enables consistent spacing in `RTL` layouts.
+
+```tsx
+<Button variant="secondary" onPress={e => submit(e)}>
+  Label
+</Button>
+```
+
+While in most cases, `children` will be a `ReactNode`, `Button` also accepts a render function with React Aria's `ButtonRenderProps`. This allows for more advanced styling and rendering options by hooking into React Aria's internal button state. You can read more about this [here](https://react-spectrum.adobe.com/react-aria/Button.html#styling).
+
+
+### Icons and positioning
+
+The `icon` property abstracts the need to handle positioning and sizing logic for icons within the `Button`. When paired with the [Icon component](/docs/illustrations-icon-icon-future-api-specification--docs), this will scale optical size and stroke width `size` prop.
+
+<Canvas of={exampleStories.ButtonWithIcon} />
+
+Set the position of the icon using the `iconPosition` prop. This will ensure content is flipped in `RTL` layouts.
+
+<Canvas of={exampleStories.ButtonWithIconEnd} />
+
+Note that icons will need the [shouldMirrorInRTL](/docs/illustrations-icon-icon-future-api-specification--docs#mirror-in-rtl) prop set to `true` when mirroring is required.
+
+
+### Icon only buttons
+
+The recommended pattern to achieve an icon only button (aka: `IconButton`) is to pass an icon in as props, then to use a [VisuallyHidden](/docs/components-visually-hidden--docs) component as `children`.
+
+<Canvas of={exampleStories.IconButton} className="mb-32"  />
+
+This pattern ensures that the `Button`'s accessible name is determined by its children, which helps to announce relevant content to the screen readers, as with the [pending state](#pending-state). You can learn more about this [accessible pattern here](https://cultureamp.atlassian.net/wiki/spaces/PA/pages/3833331831/Accessible+button+and+link+labels).
+
+### Pending state
+
+`isPending` and `pendingLabel` are used to indicate that an interaction is in progress. This will disable the `Button`'s press events while rendering a loading spinner and `pendingLabel`.
+
+<Canvas of={specStories.PendingButton} />
+
+For an icon only `Button`, this can be paired with the `isPendingLabelHidden` prop to visibly hide the `pendingLabel` to avoid layout shifts. It is important that the `pendingLabel` is still given meaningful message as this will be announced to a screen reader.
```

Comment:
> It is important that the `pendingLabel` is still given meaningful message as this will be announced to a screen reader.

Something sounds off. "should still be given a meaningful message"?

Reply:
How bout, "The `pendingLabel` should be short and declarative as this will be announced to a screen reader."?